TourMate/TourMate/Backend/Persistence/Repository/FirebaseRepository.swift

```diff
@@ -22,7 +23,7 @@ class FirebaseRepository: Repository {
 
     @MainActor
     func addItem<T: FirebaseAdaptedData>(id: String, item: T) async -> (hasAddedItem: Bool, errorMessage: String) {
-        guard Auth.auth().currentUser != nil else {
+        guard authenticationManager.getCurrentFirebaseUser() != nil else {
```

Comment:
Perhaps we can use a function like user.isLoggedIn? I think the logic whether the user is logged in should be contained within the auth manager